sympy/physics/mechanics/kane.py

```diff
@@ -704,7 +704,8 @@ def kanes_equations(self, FL, BL):
             A list of all RigidBody's and Particle's in the system.
 
         """
-
+        if len(FL) == 0:
+            FL = [(self._inertial, 0*self._inertial.x)]
```

Comment:
This fixes things, but it a hack. I'd prefer if you fixed the code that iterates through the force list and make it so that it can handle empty iterables or None. The API should be something like:

`def kanes_equation(bodies, loads=None)`

But that will require a backwards incompatible change which isn't that favorable.

But maybe we could do:

`def kanes_equation(loads=None, bodies=None):`

This would allow:

`kanes_equations(bodies=bodies)`

`kanes_equations(None, bodies)`

`kanes_equations(loads, bodies)`

`kanes_equations((, ), bodies)`

etc, which are all backwards compatible.